llama-index-core/llama_index/core/tools/query_engine.py

```diff
@@ -61,18 +61,7 @@ def metadata(self) -> ToolMetadata:
         return self._metadata
 
     def call(self, *args: Any, **kwargs: Any) -> ToolOutput:
-        if args is not None and len(args) > 0:
```

Comment:
I think these checks are still needed?

The LLM will often not write entirely correct inputs for a query engine tool

`{"query": "..."}`, `{"input": "..."}`, and so on.

A query engine only has one input (the query string) so we need to kind of validate that behaviour a bit

I suggest doing a full method override rather than relying on `super()` in the new tool you created